src/daft-algebra/src/boolean.rs

```diff
@@ -128,6 +132,40 @@ fn apply_de_morgans(expr: ExprRef) -> Transformed<ExprRef> {
     expr.transform_down(|e| Ok(transform_fn(e))).unwrap()
 }
 
+/// Check if `expr`, if used as a filter predicate, would filter out nulls if all attributes in `to_null` are set to null.
+///
+/// In other words, check if `expr` simplifies to false or null if all sub-expressions from `to_null` are set to the null literal.
+/// If this function returns true, the expression is guaranteed to be strong, however the inverse may not be true.
+pub fn predicate_removes_nulls(
```

Comment:
Some examples?
```
/// Example 1: (x > 0) AND (y > 0) would filter out nulls if x is null, since the entire expression evaluates to null.
///
/// Example 2: (x > 0) OR (y > 0) would not filter out nulls if only x is null.
///
/// Example 3: ((x > 0) OR (z > 0)) AND (y > 0) would filter out nulls if (x, z) are both null.
```